pkg/traceroute/icmp.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package traceroute
+
+import (
+	"bytes"
+	"encoding/binary"
+	"log"
+	"net"
+	"time"
+
+	"golang.org/x/net/ipv4"
+)
+
+func (t *Trace) SendTracesICMP4() {
+	conn, err := net.ListenPacket("ip4:icmp", t.src.String())
+	if err != nil {
+		log.Fatal(err)
```

Comment:
one thing -- I hope we can get log.Anything out of this package. Is there no way to return an error?